frontend/src/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import { Form, Checkbox, CheckboxProps, DropdownItemProps, Segment } from 'semantic-ui-react';
+
+interface SelectQueryBuilderProps {
+  options: DropdownItemProps[];
+  selectedOptions?: string[];
+  onUpdateOptions?: (options: string) => void;
+}
+
+const segment_style = {
```

Comment:
Stick to the naming convention

Reply:
Added 3 columns for large sources and 2 or 1 for smaller sources. Also updated naming